Core/Services/AuthenticationService.cs

```diff
@@ -73,12 +74,11 @@ public async Task<UserAutorizationDTO> LoginAsync(UserLoginDTO data)
 
         public async Task LogoutAsync(UserLogoutDTO userLogoutDTO)
         {
-            var refreshToken = _refreshTokenRepository.Query().SingleOrDefault(t => t.Token == userLogoutDTO.RefreshToken);
+            var refreshToken = (await _refreshTokenRepository
+                .FindWithSpecificationAsync(new GetRefreshTokenByToken(userLogoutDTO.RefreshToken)))
+                .First();
```

Comment:
We should NOT load the list into memory and take first element, we must load first element from db.